docs/source/distributed.fsdp.fully_shard.rst

```diff
@@ -0,0 +1,74 @@
+torch.distributed.fsdp.fully_shard
+==================================
+
+PyTorch FSDP2 (``fully_shard``)
```

Comment:
This can be done in a follow up review but we need to outline the difference between FSDP1 and FSDP2 - when would I use either of them or whether FSDP2 from now a preferred way. Possibly, add a note in the torch.distributed.fsdp that if you are starting from scratch, use FSDP2 or something along those lines.

Reply:
Are you mainly looking for the explicit recommendation of when to use FSDP1 vs. when to use FSDP2?

There is some comparison of FSDP1 vs. FSDP2 in this PR.

Reply:
I added a sentence.